### PR TITLE
Andrea part3

### PR DESCRIPTION
**Pull request by ashcreek**

### CODE REVIEW

File: src/part_03/1_questions [hunk 1]
```diff
@@ -3,64 +3,76 @@ Answer the following questions
...
-
+When you want every object created for that class to have to have the traits of that class.
 
 16) What does the finalize() method do?
-
+It removes objects from the memory when you stop referencing to them. This is called garbage collection.
 
 17) What does the "this" keyword do?
-
+"this" is used to refer to the object that you creating.
```
Comment (roioteromorales) [inline src/part_03/1_questions]: Not only when you are creating it but at point of time, just needs to be inside this class.

File: src/part_03/1_questions [hunk 2]
```diff
@@ -3,64 +3,76 @@ Answer the following questions
...
 
 18) When should you use the "this" keyword?
-
+"this" is used when creating constructors so you don't have to keep telling it what object you are talking about.
```
Comment (roioteromorales) [inline src/part_03/1_questions]: this. refers to the instance of the object itself. You should use it when you have another variable with the same name to know to which variable you are referring (the instance one or the other one)

Comment (roioteromorales) [reply src/part_03/1_questions]: Exactly as you did in the question 19, You had a parameter called the same as the field of the class, so in order to know to which one you are referring you use `this.species` to use the class one, and `species` to refer to the parameter

File: src/part_03/Exercise_05.java [hunk 3]
```diff
@@ -7,17 +7,24 @@
     public static void main(String[] strings) throws java.io.IOException {
 
         // read an integer from the user >=1 and <= 999,999,999
-
+        System.out.println("Type in a number between 1 and 999,999,999");
+        Scanner newScanner = new Scanner(System.in);
+        int inputNumber = newScanner.nextInt();
```
Comment (roioteromorales) [inline src/part_03/Exercise_05.java]: You are not checking that the number is lower than 999.999.999, You are trusting the user, never trust the user!! ;)
tenor-107567683

File: src/part_03/Exercise_06.java [hunk 4]
```diff
@@ -14,22 +14,32 @@ public static void main(String[] strings) {
...
-        // Display results using a 12 hour clock, include AM or PM
-
+        long currentHour = totalHours % 24 + timeZoneChange;
+        String amOrPm = " ";
+        if (currentHour >= 12) {
+            amOrPm = "pm";
+        } else {
+            amOrPm = "am";
+        }
+        System.out.println(currentHour + ":" + currentMinute + ":" + currentSeconds + amOrPm);
```
Comment (roioteromorales) [inline src/part_03/Exercise_06.java]: One tip, if there are `14:06:32 `then your programs display `14:6:32`,  there is a way to fix this, formatting the print use a 2 digits number

https://stackoverflow.com/questions/12421444/how-to-format-a-number-0-9-to-display-with-2-digits-its-not-a-date

File: src/part_03/1_questions (hunk 1 above)
Comment (ashcreek) [reply src/part_03/1_questions]: Oh, ya- that makes sense

**Pull request merged**